src/Microsoft.Azure.EventHubs.Processor/EventHubPartitionPump.cs

```diff
@@ -171,9 +159,11 @@ public Task ProcessEventsAsync(IEnumerable<EventData> events)
                 {
                     ProcessorEventSource.Log.PartitionPumpError(
                         this.eventHubPartitionPump.Host.Id, this.eventHubPartitionPump.PartitionContext.PartitionId, "EventHub client error:", error.ToString());
-                    await this.eventHubPartitionPump.ProcessErrorAsync(error).ConfigureAwait(false);
                 }
 
+                // We would like to deliver all errors in the pump to error handler.
+                await this.eventHubPartitionPump.ProcessErrorAsync(error).ConfigureAwait(false);
```

Comment:
>await this.eventHubPartitionPump.ProcessErrorAsync(error).ConfigureAwait(false); [](start = 16, length = 80)

We were not delivering ReceiverDisconnectedException to ProcessError callback. Let client know about this for logging and debugging purpose.